.NET/Microsoft.Recognizers.Definitions.Common/English/DateTimeDefinitions.cs

```diff
@@ -279,7 +281,9 @@ public static class DateTimeDefinitions
       public static readonly string YearPeriodRegex = $@"((((from|during|in)\s+)?{YearRegex}\s*({TillRegex})\s*{YearRegex})|(((between)\s+){YearRegex}\s*({RangeConnectorRegex})\s*{YearRegex}))";
       public static readonly string StrictTillRegex = $@"(?<till>\b(to|(un)?till?|thru|through)\b|{BaseDateTime.RangeConnectorSymbolRegex}(?!\s*(h[1-2]|q[1-4])(?!(\s+of|\s*,\s*))))";
       public static readonly string StrictRangeConnectorRegex = $@"(?<and>\b(and|through|to)\b|{BaseDateTime.RangeConnectorSymbolRegex}(?!\s*(h[1-2]|q[1-4])(?!(\s+of|\s*,\s*))))";
-      public static readonly string ComplexDatePeriodRegex = $@"(?:((from|during|in)\s+)?(?<start>.+)\s*({StrictTillRegex})\s*(?<end>.+)|((between)\s+)(?<start>.+)\s*({StrictRangeConnectorRegex})\s*(?<end>.+))";
+      public static readonly string TillRegexWithDet = $@"({TillRegex}(\s+the\b)?)";
```

Comment:
Why do we need to create these new regexes and not reuse TillRegex? Does it cause unintended side-effects?
Having these partially duplicated regexes makes maintenance harder and also further diverges from the other platforms.
It would be preferable to either treat it at the original regex, or change the extractor/parsing code to account for the article where needed.

Reply:
I was worried of possible side effects in modifying TillRegex directly. But now I have updated it and the tests do not show any issue.